### PR TITLE
Frontend components setup with state hooks

### PR DESCRIPTION
**Pull request by Bouland95**

### CODE REVIEW

**Comment by Bouland95** [comment]
@twzkraus

File: client/public/src/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const [product, setProduct] = useState({});
+  const [storeId, setStoreId] = useState(1);
+  const [stores, setStore] = useState([]);
+  const [stock, setStock] = useState([]);
+  const [colors, setColors] = useState([]);
+  const [sizes, setSizes] = useState([]);
+
+  // get the products general details such as price, name and reviews
+  const getProduct = async (productId) => {
+    const response = await axios.get(`http://localhost:3000/products/${productId}`);
```
Comment (twzkraus) [inline client/public/src/components/app.jsx:19]: if you have this running through your server, I think you can trim it down to just `/products/${productId}`

File: client/public/src/components/app.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import axios from 'axios';
+
+import Details from './details.jsx';
+import Options from './options.jsx';
+
+const App = () => {
+  const [productId, setProductId] = useState(1);
```
Comment (twzkraus) [inline client/public/src/components/app.jsx:9]: I wonder if you can simplify this a little and remove the `productId` and `storeId` states? It doesn't look like these change anywhere--could you parse them out of the `product` and `store` variables when needed?

File: client/public/src/components/app.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  const [product, setProduct] = useState({});
+  const [storeId, setStoreId] = useState(1);
+  const [stores, setStore] = useState([]);
```
Comment (twzkraus) [inline client/public/src/components/app.jsx:12]: I think it's best practice to have these be consistent (`store` vs. `stores`)

File: client/public/src/components/app.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  const getStores = async () => {
+    const response = await axios.get(`http://localhost:3000/stores`);
+    const storesData = await response.data;
+    setStore(storesData);
+  }
+
+  // get all available stock for the selected item in the selected store
+  const getStock = async (productId, stores) => {
+    const response = await axios.get(`http://localhost:3000/stock/${productId}`);
+    const stockData = await response.data;
```
Comment (twzkraus) [inline client/public/src/components/app.jsx:34]: this step shouldn't be asynchronous--I don't think `await` is necessary

File: client/public/src/components/app.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    // find the store that matches the selected store state
+    const selectedStore = await stores.find(store => store.id === storeId);
+    if (selectedStore !== undefined) {
+      // filter to only have the selected store's stock for the selected product
+      const productStock = await stockData.filter(item => item.location === selectedStore.location);
```
Comment (twzkraus) [inline client/public/src/components/app.jsx:39]: also not asynchronous

File: client/public/src/components/details.jsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React, {Component} from 'react';
```
Comment (twzkraus) [inline client/public/src/components/details.jsx:1]: you probably don't need Component in this or any of the functional components

**Review by twzkraus — COMMENTED** [review]
I have a couple of suggestions about your states and async/await, but overall it looks good!

File: client/public/src/components/app.jsx (hunk 1 above)
Comment (Bouland95) [reply client/public/src/components/app.jsx:19]: oh that's right!

File: client/public/src/components/app.jsx (hunk 2 above)
Comment (Bouland95) [reply client/public/src/components/app.jsx:9]: These are the selected product, and selected store, but I changed the name to make it shorter. I figured if I were to make it possible to switch stores or product, I would have to keep them in a state.

File: client/public/src/components/app.jsx (hunk 3 above)
Comment (Bouland95) [reply client/public/src/components/app.jsx:12]: I made it so things that are plural have a collection of things such as stores having all the stores, while things like product have a single item, the selected product.

File: client/public/src/components/details.jsx (hunk 6 above)
Comment (Bouland95) [reply client/public/src/components/details.jsx:1]: good catch!

File: client/public/src/components/app.jsx (hunk 3 above)
Comment (Bouland95) [reply client/public/src/components/app.jsx:12]: oh nevermind I see what you mean, stores, and setStore. Thanks!

**Review by Bouland95 — COMMENTED** [review]
@twzkraus Made the changes, can you approve them?

**Comment by twzkraus** [comment]
Cool, the changes look good to me.

**Pull request merged**